js_modules/dagit/src/execute/ConfigEditorPresetsPicker.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from "react";
+import { isEqual } from "lodash";
+import { Button, Menu } from "@blueprintjs/core";
+import { Select } from "@blueprintjs/select";
+import { Query, QueryResult } from "react-apollo";
+import {
+  ConfigPresetsQuery,
+  ConfigPresetsQuery_presetsForPipeline
+} from "./types/ConfigPresetsQuery";
+import gql from "graphql-tag";
+import { IExecutionSession } from "../LocalStorage";
+
+type Preset = ConfigPresetsQuery_presetsForPipeline;
+
+const PresetSelect = Select.ofType<Preset>();
+
+const PresetDivider: Preset = {
+  __typename: "PipelinePreset",
+  name: "divider",
+  solidSubset: null,
+  environment: null
+};
+
+interface ConfigEditorPresetsPickerProps {
+  pipelineName: string;
+  solidSubset: string[] | null;
+  onCreateSession: (initial: Partial<IExecutionSession>) => void;
+}
+
+export default class ConfigEditorPresetsPicker extends React.Component<
+  ConfigEditorPresetsPickerProps
+> {
+  render() {
+    const { pipelineName, solidSubset, onCreateSession } = this.props;
+
+    return (
+      <Query
+        query={CONFIG_PRESETS_QUERY}
+        fetchPolicy="cache-and-network"
+        variables={{ pipelineName }}
+      >
+        {({ data }: QueryResult<ConfigPresetsQuery, any>) => {
+          const presets = ((data && data.presetsForPipeline) || []).sort(
+            (a, b) => a.name.localeCompare(b.name)
+          );
+
+          let grouped = presets.filter(p =>
+            isEqual(p.solidSubset, solidSubset)
+          );
+          if (grouped.length && grouped.length !== presets.length) {
+            grouped.push(PresetDivider);
+          }
+          grouped.push(
+            ...presets.filter(p => !isEqual(p.solidSubset, solidSubset))
+          );
```

Comment:
i think we can drop the divider and sorting since we are just applying the new state in a new workplace

Reply:
Sounds good - just pushed up a change removing this. 👍